src/range.rs

```diff
@@ -364,28 +465,59 @@ fn interval_to_string<V: Version>((start, maybe_end): &Interval<V>) -> String {
 pub mod tests {
     use proptest::prelude::*;
 
-    use crate::version::NumberVersion;
-
     use super::*;
 
-    pub fn strategy() -> impl Strategy<Value = Range<NumberVersion>> {
-        prop::collection::vec(any::<u32>(), 0..10).prop_map(|mut vec| {
-            vec.sort_unstable();
-            vec.dedup();
-            let mut pair_iter = vec.chunks_exact(2);
-            let mut segments = SmallVec::empty();
-            while let Some([v1, v2]) = pair_iter.next() {
-                segments.push((NumberVersion(*v1), Some(NumberVersion(*v2))));
-            }
-            if let [v] = pair_iter.remainder() {
-                segments.push((NumberVersion(*v), None));
-            }
-            Range { segments }
-        })
+    pub fn strategy() -> impl Strategy<Value = Range<u32>> {
+        (
+            any::<(bool, bool)>(),
+            prop::collection::vec(any::<(u32, bool)>(), 1..10),
+        )
+            .prop_map(|((start_bounded, end_bounded), mut vec)| {
+                // Ensure the bounds are increasing and non-repeating
+                vec.sort_by_key(|(value, _)| *value);
+                vec.dedup_by_key(|(value, _)| *value);
```

Comment:
Thinking about this more I'm now realizing that this cannot generate `[(Unbounded, Excluded(1)), (Excluded(1), Unbounded)]`. But I'm not seeing an obvious way to fix it.

Reply:
Here is an alternative strategy https://github.com/pubgrub-rs/pubgrub/commit/ad389723cc9e2f69618923c6114ab59e32c1ccba thoughts?

Reply:
I tried rewriting the strategy based on your proposal (based on @Eh2406 code) and found that it's basically almost what you describe @mpizenberg . I've committed a version that includes comments (c44639a02e08c3065771c80a3905b36af17d131e).